Extension/src/Debugger/configurationProvider.ts

```diff
@@ -467,22 +476,23 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
                         }
                         const compilerDirname: string = path.dirname(compilerPath);
                         const debuggerPath: string = path.join(compilerDirname, debuggerName);
-                        if (isWindows) {
-                            newConfig.miDebuggerPath = debuggerPath;
-                            return resolve(newConfig);
-                        } else {
-                            fs.stat(debuggerPath, (err, stats: fs.Stats) => {
-                                if (!err && stats && stats.isFile()) {
-                                    newConfig.miDebuggerPath = debuggerPath;
-                                } else {
-                                    newConfig.miDebuggerPath = path.join("/usr", "bin", debuggerName);
+                        if (!await util.checkFileExists(debuggerPath)) {
```

Comment:
Could you clarify why it checks `await util.checkFileExists(debuggerPath)` as false at line 479 and then at line 481 it checks it again as true when there doesn't seem to be a change to `debuggerPath` that is being verified? 

It seems conditional block 479 could be refactored?

Reply:
line 481 had to use `newDebuggerPath`, it is fixed now.